platform/ios/platform/ios/src/MLNMapView.mm

```diff
@@ -4376,6 +4387,9 @@ - (MLNMapCamera *)cameraForCameraOptions:(const mbgl::CameraOptions &)cameraOpti
     {
         options.pitch = camera.pitch;
     }
+    if (!MGLEdgeInsetsEqual(camera.padding, MGLEdgeInsetsZero)) {
+        options.padding = MLNEdgeInsetsFromNSEdgeInsets(camera.padding);
+    }
```

Comment:
Moving the discussion of the old PR over to this PR:

>Not sure if this is correct, as there is edgePadding as argument as well as in padding in MGLMapCamera. The padding in MGLMapCamera overwrites the param here.

Reply:
Let me do some more testing before we merge this.

Reply:
I did my testing. However, since this is my first PR in maplibre native I would like to have feedback from someone that knows this part of the code well enough :)

Reply:
> Not sure if this is correct, as there is edgePadding as argument as well as in padding in MGLMapCamera. The padding in MGLMapCamera overwrites the param here.

Unfortunately, the library currently conflates two different concepts as “padding”:

* `MLNMapView.contentInset` is supposed to be persistent, something the developer sets according to UI that takes up space within the map view, such as a translucent sidebar or a camera “notch” on iOS. The developer should only have to think about this inset when sizing the map view or embedding a subview, not when manipulating or animating the camera. The resulting virtual rectangle is often called a “safe area” on Apple platforms.
* Various camera-manipulating methods like `-[MLNMapView setCamera:withDuration:animationTimingFunction:edgePadding:completionHandler:]` take an `edgePadding` parameter for more precisely targeting the camera transition. In particular, the navigation SDK (and navigation applications in general) need to anchor the user location indicator somewhere other than the center of the safe area, typically in the lower third of the map view.

In the original design in mapbox/mapbox-gl-native#3583, the `mbgl::CameraOptions::padding` was the sum of the (persistent) content insets and the (temporary) edge padding. `mbgl::CameraOptions::padding` affected the frame of reference for `mbgl::CameraOptions::center`, but `mbgl::Map::transform` never persisted the padding because it had no context about how much of that padding was persistent and how much was temporary.

Unfortunately, a change was made to mbgl that conflated these two concepts, creating quite a headache for iOS/macOS developers: mapbox/mapbox-gl-native#15232 mapbox/mapbox-gl-native#15233. I have recommended that `mbgl::Transform` decouple the concepts of “padding” and “insets” and even replace “padding” with a more explicit way to anchor the center coordinate within screen space: https://github.com/mapbox/mapbox-gl-native/issues/15233#issuecomment-621507580.